src/main/java/org/spongepowered/api/service/ServiceProvider.java

```diff
@@ -0,0 +1,86 @@
+package org.spongepowered.api.service;
+
+import org.spongepowered.api.service.ban.BanService;
+import org.spongepowered.api.service.economy.EconomyService;
+import org.spongepowered.api.service.pagination.PaginationService;
+import org.spongepowered.api.service.permission.PermissionService;
+import org.spongepowered.api.service.rcon.RconService;
+import org.spongepowered.api.service.sql.SqlService;
+import org.spongepowered.api.service.user.UserStorageService;
+import org.spongepowered.api.service.whitelist.WhitelistService;
+
+import java.util.Optional;
+
+/**
+ * Provides various Sponge services.
+ */
+public interface ServiceProvider {
+
+    /**
+     * Provides the service represented by the supplied {@link Class}.
+     *
+     * @param serviceClass The class
+     * @param <T> The type of service
+     * @return The service, if one exists
+     */
+    <T> Optional<T> provide(Class<T> serviceClass);
+
+    /**
+     * Provides the {@link ServiceRegistration} for the supplied {@link Class}.
+     *
+     * @param serviceClass The class
+     * @param <T> The type of service
+     * @return The registration, if one exists
+     */
+    <T> Optional<ServiceRegistration<T>> registration(Class<T> serviceClass);
```

Comment:
I would call it `getRegistration`, not sure which prupose this method has though.

Reply:
Yeah, name is in the air. A example is that a plugin might want to know who supplies a service, so they then know they can use an expanded API from said plugin that they are aware of.